server/src/main/java/com/cloud/resource/ResourceManagerImpl.java

```diff
@@ -1286,12 +1288,31 @@ public Host maintain(final PrepareForMaintenanceCmd cmd) {
             throw new InvalidParameterValueException("Unable to find host with ID: " + hostId + ". Please specify a valid host ID.");
         }
 
-        if (_hostDao.countBy(host.getClusterId(), ResourceState.PrepareForMaintenance, ResourceState.ErrorInMaintenance) > 0) {
-            throw new InvalidParameterValueException("There are other servers in PrepareForMaintenance OR ErrorInMaintenance STATUS in cluster " + host.getClusterId());
+        final ResourceState hostState = host.getResourceState();
+        if (hostState == ResourceState.Maintenance || hostState == ResourceState.PrepareForMaintenance ||
+                hostState == ResourceState.ErrorInPrepareForMaintenance) {
+            throw new CloudRuntimeException("Host is already in state " + hostState + ". Cannot recall for maintenance until resolved.");
+        }
+
+        if (_hostDao.countBy(host.getClusterId(), ResourceState.PrepareForMaintenance, ResourceState.ErrorInPrepareForMaintenance) > 0) {
```

Comment:
Should this be `> 1`, just check if we should not consider the specific host already in maintenace etc?

Reply:
I've checked that in the if above. Also > 1 is a good point but that would break the existing behaviour which shuts VMs for single host. We have created an open github issue to discuss and change that.